Python/Include/VariableTraits.hpp

```diff
@@ -0,0 +1,59 @@
+#ifndef ROVER_PYTHON_VARIABLE_TRAITS_HPP
+#define ROVER_PYTHON_VARIABLE_TRAITS_HPP
+#include <pybind11/pybind11.h>
+#include "Rover/VariableTraits.hpp"
+#include "Arithmetics.hpp"
+
+namespace Rover {
+
+  //! A struct wrapper to check whether a python object is continuous for
+  //! modeling purposes for a given scalar type.
+  /*!
+    \tparam S The type of the scalar.
+  */
+  template<typename S>
+  struct ContinuousChecker<S, pybind11::object> {
+
+    //! Checks whether a python object is continuous for modeling purposes for
+    //! scalar S.
+    static bool check(const pybind11::object& obj);
+  };
+
+  template<typename S>
+  bool ContinuousChecker<S, pybind11::object>::check(const pybind11::object&
+      obj) try {
+    {
+      auto val = obj + obj;
+    }
+    {
+      auto val = -obj;
+    }
+    {
+      auto val = obj == obj;
+    }
+    {
+      auto val = obj != obj;
+    }
+    {
+      auto val = obj < obj;
+    }
+    {
+      auto val = S() * obj;
+    }
+    try {
+      auto val = obj / obj;
+    } catch(pybind11::error_already_set& e) {
+      e.restore();
+      if(PyErr_ExceptionMatches(PyExc_ZeroDivisionError)) {
+        PyErr_Clear();
+      } else {
+        throw pybind11::error_already_set();
```

Comment:
Use ```throw;``` to rethrow the exception.

Reply:
`error_already_set` constructor fetches the exception from Python C API, which has intercepted the original exception due to `restore()`, so it is the only way to correctly rethrow the context of the original exception. In this case though, it does not really matter as the exception is not processed.

Reply:
Okay thanks.